engine/apps/user_management/tests/test_sync.py

```diff
@@ -111,42 +112,54 @@ def test_sync_organization(make_organization, make_team, make_user_for_organizat
             "role": "admin",
             "avatarUrl": "test.test/test",
             "permissions": [],
+        }
+    ],
+)
+@patch.object(
+    GrafanaAPIClient,
+    "get_teams",
+    return_value=(
+        {
+            "totalCount": 1,
+            "teams": (
+                {
+                    "id": 1,
+                    "name": "Test",
+                    "email": "test@test.test",
+                    "avatarUrl": "test.test/test",
+                },
+            ),
         },
-    )
-
-    api_teams_response = {
-        "totalCount": 1,
-        "teams": (
-            {
-                "id": 1,
-                "name": "Test",
-                "email": "test@test.test",
-                "avatarUrl": "test.test/test",
-            },
-        ),
-    }
+        None,
+    ),
+)
+@patch.object(GrafanaAPIClient, "check_token", return_value=(None, {"connected": True}))
+@patch.object(GrafanaAPIClient, "get_grafana_plugin_settings", return_value=({"enabled": True}, None))
+@patch("apps.user_management.sync.org_sync_signal")
+def test_sync_organization(
+    mocked_org_sync_signal,
+    _mock_get_grafana_plugin_settings,
+    _mock_check_token,
+    _mock_get_teams,
+    _mock_get_users,
+    _mock_is_rbac_enabled_for_organization,
+    make_organization,
+):
+    organization = make_organization()
 
     api_members_response = (
-        {
-            "orgId": organization.org_id,
-            "teamId": 1,
-            "userId": 1,
-        },
+        [
+            {
+                "orgId": organization.org_id,
+                "teamId": 1,
+                "userId": 1,
+            }
+        ],
+        None,
     )
 
-    api_check_token_call_status = {"connected": True}
-
-    with patch.object(GrafanaAPIClient, "is_rbac_enabled_for_organization", return_value=False):
-        with patch.object(GrafanaAPIClient, "get_users", return_value=api_users_response):
-            with patch.object(GrafanaAPIClient, "get_teams", return_value=(api_teams_response, None)):
-                with patch.object(GrafanaAPIClient, "get_team_members", return_value=(api_members_response, None)):
-                    with patch.object(
-                        GrafanaAPIClient, "check_token", return_value=(None, api_check_token_call_status)
-                    ):
-                        with patch.object(
-                            GrafanaAPIClient, "get_grafana_plugin_settings", return_value=({"enabled": True}, None)
-                        ):
```

Comment:
mostly just refactored this for readability sake.. having 8 nested context managers is really difficult to deal with...